language-support/java/codegen/src/ledger-tests/scala/com/digitalasset/CodegenTest.scala

```diff
@@ -0,0 +1,8 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml
+
+import org.scalatest.flatspec.AsyncFlatSpec
+
+class CodegenTest extends AsyncFlatSpec with LedgerTest with InterfacesTest with StakeholdersTest
```

Comment:
@samuel-williams-da you could factorize the 3 tests in https://github.com/digital-asset/daml/pull/16763 like I have done here. 
You only need to make `LedgerTest`, `InterfacTest` and `StackholderTest` `trait`